lib/publiccloud/provider.pm

```diff
@@ -705,6 +705,25 @@ sub terraform_param_tags
     return encode_json($tags);
 }
 
+=head2 get_terraform_output
+
+Query the terraform data structure in json format.
+Input: <jq-query-format> string; <empty> = no query then full output of data structure.
+E.g: to get the VM instance name from json data structure, the call is: 
+    get_terraform_output(".vm_name.value[0]");
+To get the complete output structure, the call is:
+    get_terraform_output();
+
+=cut
+
+sub get_terraform_output {
+    my ($self, $jq_query) = @_;
+    assert_script_run('cd ~/terraform');
```

Comment:
If SAP will still need this publiccloud code-flow, ok I will try to align it

Reply:
Applying this customization open the door to SLES4SAP to use it. 
It could be useful in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/42d9cf281e7a021a04d9583293ad75f8355cea1a/lib/sles4sap_publiccloud.pm#L726

This function at the moment is implemented reading the Ansible inventory but I think terraform output has the same info too.

All in all, hardcoded path aside, this function is doing a real generic task: 
- run `terraform output`
- pipe the output through `jq` with a jq query string provided by the caller
- return the result as string

Anyway totally up to you

Reply:
Well, if SAP does **not using** anywhere the `gce->cleanup` routine here updated, the update proposal could eventually part of a new dedicated PR, if ok to you.

But if SAP does use it, change needed now to align the usage.

Reply:
@mpagot Does SAP already have a setting for this or how would you suggest to have this done?

Reply:
> SAP does not using anywhere the gce->cleanup 
This is true, no any SLES4SAP use that cleanup function.

> Does SAP already have a setting for this or how would you suggest to have this done?
This PR as it is now does not require any change in any existing SAP test
My comment was just an abstract and generic idea looking at the code of this particular function, out of any context, not considering the `cleanup()` caller or anything. Looking at the code all parts are looking generic with no dependency on any QE-C or QE-SAP specific deployment strategy and folder organization. From some point of view it can be described as "a perl function calling `terraform output` and extracting some string from its output using a jq query provided by the caller"
The only not generic part in it is the `cd ~/teraform`. Generally speaking it is not generic, it suppose that the terraform state files are exactly and only in a specific folder.

This PR can be merged as it is right now from my point of view.

Reply:
Thank you much Michele for your clarification. Then we proceed with the current status.

Eventually, after merged, we can deal with improvements to the `get` routine, like on-demand customized work-directory, useful for SAP too.

Reply:
Something to check about the terraform path https://openqa.suse.de/tests/14598491#step/ssh_interactive_end/32 (does not means this PR is wrong)